src/BlazorWebView/src/Microsoft.AspNetCore.Components.WebView.Maui/Android/BlazorWebViewHandler.Android.cs

```diff
@@ -0,0 +1,133 @@
+			}
+
+			_webViewClient = GetWebViewClient();
+			aWebView.SetWebViewClient(_webViewClient);
+
+			_webChromeClient = GetWebChromeClient();
+			aWebView.SetWebChromeClient(_webChromeClient);
+
+			return aWebView;
+		}
+
+		protected override void DisconnectHandler(AWebView nativeView)
+		{
+			nativeView.StopLoading();
+
+			_webViewClient?.Dispose();
+			_webChromeClient?.Dispose();
+		}
+
+		private bool RequiredStartupPropertiesSet =>
+			//_webview != null &&
+			HostPage != null &&
+			Services != null;
+
+		public string? HostPage { get; private set; }
+		public ObservableCollection<RootComponent>? RootComponents { get; private set; }
+		public new IServiceProvider? Services { get; private set; }
+
+		private void StartWebViewCoreIfPossible()
+		{
+			if (!RequiredStartupPropertiesSet ||
+				false)//_webviewManager != null)
+			{
+				return;
+			}
+			if (TypedNativeView == null)
+			{
+				throw new InvalidOperationException($"Can't start {nameof(BlazorWebView)} without native web view instance.");
+			}
+
+			var resourceAssembly = RootComponents?[0]?.ComponentType?.Assembly;
+			if (resourceAssembly == null)
+			{
+				throw new InvalidOperationException($"Can't start {nameof(BlazorWebView)} without a component type assembly.");
+			}
+
+			// We assume the host page is always in the root of the content directory, because it's
+			// unclear there's any other use case. We can add more options later if so.
+			var contentRootDir = Path.GetDirectoryName(HostPage) ?? string.Empty;
+			var hostPageRelativePath = Path.GetRelativePath(contentRootDir, HostPage!);
+			var fileProvider = new ManifestEmbeddedFileProvider(resourceAssembly, root: contentRootDir);
+
+			_webviewManager = new AndroidWebKitWebViewManager(this, TypedNativeView, Services!, MauiDispatcher.Instance, fileProvider, hostPageRelativePath);
+			if (RootComponents != null)
+			{
+				foreach (var rootComponent in RootComponents)
+				{
+					// Since the page isn't loaded yet, this will always complete synchronously
+					_ = rootComponent.AddToWebViewManagerAsync(_webviewManager);
+				}
+			}
+
+			_webviewManager.Navigate("/");
+		}
+
+		public static void MapHostPage(BlazorWebViewHandler handler, IBlazorWebView webView)
+		{
+			handler.HostPage = webView.HostPage;
+			handler.StartWebViewCoreIfPossible();
+		}
+
+		public static void MapRootComponents(BlazorWebViewHandler handler, IBlazorWebView webView)
+		{
+			handler.RootComponents = webView.RootComponents;
+			handler.StartWebViewCoreIfPossible();
+		}
+
+		public static void MapServices(BlazorWebViewHandler handler, IBlazorWebView webView)
+		{
+			handler.Services = webView.Services;
+			handler.StartWebViewCoreIfPossible();
+		}
+
+		public void LoadHtml(string? html, string? baseUrl)
+		{
+			TypedNativeView?.LoadDataWithBaseURL(baseUrl ?? AssetBaseUrl, html ?? string.Empty, "text/html", "UTF-8", null);
+		}
+
+		public void LoadUrl(string? url)
+		{
+			TypedNativeView?.LoadUrl(url ?? string.Empty);
+		}
```

Comment:
Since I don't really know MAUI, I can't really give super useful comments about this, but I do wonder about all these `public` methods, doubly so the `public static` ones. Is the public API surface for consumers really that broad?

Is this just temporary, or an expected pattern within MAUI?

Reply:
Some of this is temporary (I think this specific method will get deleted). But for some others, it's part of the MAUI pattern. Once I have everything running and cleaned up, including code comments, I think most things will become clearer.